msteams-platform/tabs/how-to/build-adaptive-card-tabs.md

```diff
@@ -0,0 +1,375 @@
+{
+  "$schema": "https://raw.githubusercontent.com/OfficeDev/microsoft-teams-app-schema/preview/DevPreview/MicrosoftTeams.schema.json",
+  "manifestVersion": "devPreview",
+  "id": "4b0b7c70-d2bc-11ea-a813-bfc190db5187",
+  "version": "0.0.1",
+  "packageName": "acprototype",
+  "developer": {
+    "name": "proto",
+    "websiteUrl": "https://acprototype.azurewebsites.net",
+    "privacyUrl": "https://acprototype.azurewebsites.net/privacy.html",
+    "termsOfUseUrl": "https://acprototype.azurewebsites.net/tou.html"
+  },
+  "name": {
+    "short": "Workday",
+    "full": "Workday Prototype"
+  },
+  "description": {
+    "short": "TODO: add short description here",
+    "full": "TODO: add full description here"
+  },
+  "icons": {
+    "outline": "icon-outline.png",
+    "color": "icon-color.png"
+  },
+  "accentColor": "#D85028",
+  "configurableTabs": [],
+  "staticTabs": [
+    {
+      "entityId": "workday",
+      "name": "Home",
+      "contentBotId": "eda13c8b-ec36-4ef5-a600-999c9531a536",
+      "scopes": ["personal"]
+    },
+    {
+      "entityId": "samples",
+      "name": "More",
+      "contentBotId": "eda13c8b-ec36-4ef5-a600-999c9531a536",
+      "scopes": ["personal"]
+    }
+  ],
+  "connectors": [],
+  "composeExtensions": [],
+  "permissions": ["identity", "messageTeamMembers"],
+  "validDomains": [
+    "andhillo-relay.servicebus.windows.net",
+    "acprototype.azurewebsites.net",
+    "token.botframework.com"
+  ]
+}
+```
+
+## Invoke activities
+
+Communication between your Adaptive Card Tab and your bot is done through `invoke` activities. Each `invoke` activity has a corresponding *name*. Use the name of each activity to differentiate each request. `tab/fetch` and `tab/submit` are the activities covered in this section.
+
+### Fetch Adaptive Card to render to a tab
+
+`tab/fetch` is the first invoke request that your bot receives when a user opens an Adaptive Card Tabs. When your bot receives the request, it will either send a tab **continue** response or a tab **auth** response.
+The **continue** response includes an array for **cards**, which is rendered vertically to the tab in the order of the array.
+
+The following image is an example of the **continue** response, where each card is an Adaptive Card:
+
+:::image type="content" source="../../assets/images/tabs/adaptive-cards-rendered-in-tabs.png" alt-text="Example of Adaptive Card rendered in tabs." border="false":::
+
+> [!NOTE]
+> The **auth** response is explained in detail in the [authentication](#authentication) section.
+
+The following code snippets are examples of `tab/fetch` request and response:
+
+**`tab/fetch` request**
+
+```json
+// tab/fetch POST request: agents/{botId}/invoke
+{
+    "name": "tab/fetch",
+    "value: {
+        "tabContext": {
+            "tabEntityId": "{tab_entity_id}"
+        },
+        "context": {
+            "theme": "default"
+            }
+    },
+    "conversation": {
+        "id": "tab:hash(tabId_appId_userId)"
```

Comment:
I don't think we want to share how this conversation id is generated with the public. It's probably best to keep this obfuscated, something like tab:conversationId? or just "generated conversation id"

Reply:
updated to "generated conversation id"